### PR TITLE
Update to sigterm instead of sigkill for Linux

### PR DESCRIPTION
**Pull request by sanfern**
Signed-off-by: Santhosh Fernandes <santhosh.fernandes@gmail.com>

### CODE REVIEW

**Review by satyaranjanp — APPROVED** [review]
LGTM

File: kf/bpf.go [hunk 1]
```diff
@@ -227,8 +227,8 @@ func (b *BPF) Stop(ifaceName, direction string, chain bool) error {
 	stats.Set(0.0, stats.NFRunning, b.Program.Name, direction)
 
 	if len(b.Program.CmdStop) < 1 {
-		if err := b.Cmd.Process.Kill(); err != nil {
-			return fmt.Errorf("BPFProgram %s kill failed with error: %w", b.Program.Name, err)
+		if err := b.ProcessTerminate(); err != nil {
```
Comment (dthaler) [inline kf/bpf.go:230]: is there no cross-plat b.Cmd.Process.Terminate() ?

File: kf/kf_unix.go [hunk 2]
```diff
@@ -87,6 +87,14 @@ func (b *BPF) SetPrLimits() error {
 	return nil
 }
 
+// ProcessTerminate - Send sigterm to the process
+func (b *BPF) ProcessTerminate() error {
+	if err := syscall.Kill(b.Cmd.Process.Pid, syscall.SIGTERM); err != nil {
```
Comment (dthaler) [inline kf/kf_unix.go:92]: can you do `b.Cmd.Process.Signal(syscall.SIGTERM)` or will that only work for SIGINT and SIGKILL?

**Review by dthaler — APPROVED** [review]

File: kf/bpf.go [hunk 3]
```diff
@@ -23,6 +23,7 @@ import (
 	"strconv"
 	"strings"
 	"sync"
+	"syscall"
```
Comment (dthaler) [inline kf/bpf.go:26]: can line 26 be removed now?

**Review by dthaler — APPROVED** [review]

File: kf/bpf.go (hunk 3 above)
Comment (sanfern) [reply kf/bpf.go:26]: We are using syscall here - line# 231 b.Cmd.Process.Signal(syscall.SIGTERM)

**Review by jniesz — APPROVED** [review]
LGTM

**Pull request merged**